Cannot find type definition file for ‘zen-observable'

### Description

I'm getting these errors when building the package.

```
node_modules/mock-apollo-client/dist/mockSubscription.d.ts:1:23 - error TS2688: Cannot find type definition file for 'zen-observable'.

1 /// <reference types="zen-observable" />
                        ~~~~~~~~~~~~~~

node_modules/mock-apollo-client/dist/mockSubscription.d.ts:17:25 - error TS2503: Cannot find namespace 'ZenObservable'.

17     subscribe(observer: ZenObservable.SubscriptionObserver<FetchResult>): void;
```

### Environment

@apollo/client: "^3.6.6"
mock-apollo-client: "^1.2.0"

### Investigation

ZenObservable.Subscription has been replaced with ObservableSubscription in @apollo/client: 3.5.0 (https://github.com/apollographql/apollo-client/pull/8695)

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -19,6 +19,6 @@ jobs:
         node-version: 16
     - run: npm ci
     - run: npm run build
-    - run: npm test
+    - run: npm run test:suite
       env:
         CI: true

--- a/jest.config.js
+++ b/jest.config.js
@@ -1,3 +1,10 @@
+const displayName = process.env.JEST_DISPLAY_NAME;
+
+const additionalConfig = displayName
+  ? { displayName }
+  : {};
+
 module.exports = {
   preset: 'ts-jest',
+  ...additionalConfig,
 };

--- a/package.json
+++ b/package.json
@@ -8,6 +8,7 @@
     "build": "rm -rf ./dist && tsc",
     "test": "jest",
     "test:watch": "jest --watch",
+    "test:suite": "./runTestSuite.sh",
     "deploy": "npm run test && npm run build && npm publish"
   },
   "repository": {

--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
   },
   "homepage": "https://github.com/mike-gibson/mock-apollo-client#readme",
   "devDependencies": {
-    "@apollo/client": "^3.1.3",
+    "@apollo/client": "^3.6.9",
     "@types/jest": "^28.1.6",
     "@types/node": "^16.11.46",
     "graphql": "^15.3.0",

--- a/runTestSuite.sh
+++ b/runTestSuite.sh
@@ -1,0 +1,39 @@
+#!/bin/bash
+
+# This script loops through a list of apollo client npm package versions
+# and runs the jest tests for each version.
+
+apolloVersions=(
+  "@apollo/client@3.0.2"
+  "@apollo/client@3.1.5"
+  "@apollo/client@3.2.9"
+  "@apollo/client@3.3.21"
+  "@apollo/client@3.4.17"
+  "@apollo/client@3.5.10"
+  "@apollo/client@3.6.9"
+)
+exitStatus=0
+
+for apolloVersion in "${apolloVersions[@]}"
+do
+  echo "Running tests for $apolloVersion"
+
+  npm install --no-save $apolloVersion
+  npmExitCode=$?
+
+  if [ $npmExitCode -ne 0 ]; then
+    echo "npm install failed, exiting"
+    exitStatus=99
+    break
+  fi
+
+  export JEST_DISPLAY_NAME=$apolloVersion
+  npm test -- || exitStatus=$?
+done
+
+# Revert changes to node_modules folder
+npm install
+
+echo "Finished"
+
+exit $exitStatus

--- a/src/mockSubscription.test.ts
+++ b/src/mockSubscription.test.ts
@@ -1,7 +1,7 @@
 import { FetchResult } from '@apollo/client/core';
-import { MockSubscription } from './mockSubscription';
+import { MockSubscription, SubscriptionObserver } from './mockSubscription';
 
-class MockObserver implements ZenObservable.SubscriptionObserver<FetchResult> {
+class MockObserver implements SubscriptionObserver<FetchResult> {
   closed: boolean;
   next: (value: FetchResult) => void;
   error: (errorValue: any) => void;

--- a/src/mockSubscription.ts
+++ b/src/mockSubscription.ts
@@ -9,18 +9,29 @@ export interface IMockSubscription<TData = any> {
 }
 
 export type MockSubscriptionOptions = {
-  disableLogging?: boolean
+  disableLogging?: boolean;
+}
+
+/**
+ * SubscriptionObserver interface copied from zen-observable-ts in order to be compatible
+ * with @apollo/client library pre-3.5 which used zen-observable and different imports/types
+ */
+ export interface SubscriptionObserver<T> {
+  closed: boolean;
+  next(value: T): void;
+  error(errorValue: any): void;
+  complete(): void;
 }
 
 export class MockSubscription<TData = any> implements IMockSubscription<TData> {
-  private observer?: ZenObservable.SubscriptionObserver<FetchResult>;
+  private observer?: SubscriptionObserver<FetchResult>;
   private loggingDisabled: boolean;
 
   constructor(options?: MockSubscriptionOptions) {
     this.loggingDisabled = options?.disableLogging ?? false;
   }
 
-  subscribe(observer: ZenObservable.SubscriptionObserver<FetchResult>) {
+  subscribe(observer: SubscriptionObserver<FetchResult>) {
     if (this.observer && !this.loggingDisabled) {
       console.warn(
         'Warning: mock-apollo-client - Mock subscription was already being used for a previous query. ' +

